src/Tgstation.Server.Host/Components/Deployment/DreamMaker.cs

```diff
@@ -340,9 +340,11 @@ await databaseContextFactory.UseContext(
 					repositorySettings!.PushTestMergeCommits!.Value
 					&& repositorySettings.AccessToken != null
 					&& repositorySettings.AccessUser != null;
+				var oldCompileJob = await compileJobConsumer.LatestCompileJob();
 				using (repo)
```

Comment:
Noticing a bug here, Can you move the declaration of `likelyPushedTestMergeCommit` and `oldCompileJob` into the scope of `using (repo)`